### ISSUE TITLE
Make the documentation follow the common template

### ISSUE THREAD

**Issue by kevinsimper** [craftyjs/Crafty#534]
Instead of applying the template, just inserting this instead:

```
--- 
layout: default

---
```

This will make the craftyjs.com website easier to maintain, since one edit in the template will reflect the whole website, and not like now were the API documentation is very static.

**Comment by starwed** [comment]
I might be confused, but I think this removes the ability to build the docs locally.

Can we hook grunt up to a jekyl processor, so that it also performs that step of generating the documentation?

**Comment by kevinsimper** [comment]
I can see what you mean.

It is not possible to do with Github Pages, sadly. Then we have to move the website to a other host.

I think the best solution would be to create two different grunt process, one for local api documentation and one for the craftyjs.com website.

**Comment by starwed** [comment]
> It is not possible to do with Github Pages, sadly. Then we have to move the website to a other host.

I was wondering if we could locally use a similar processor to the one Pages uses, to generate the files locally.

Alternatively, we could just not use Jekyl in favor of some other approach -- nothing about Github Pages prevents this.  :)

**Comment by kevinsimper** [comment]
We could something like https://github.com/visionmedia/dox which makes the documentation into a json file, and then we could use whatever template system.

I would really like that it would be a one-page documentation, because it is that much faster to jump between and you can search the whole documentation with the native browser search.

**Comment by starwed** [comment]
> We could something like https://github.com/visionmedia/dox which makes the documentation into a json file, and then we could use whatever template system.

There are two separate issues here: how we parse documentation from the code, and how we generate a website that hosts not just auto-generated content, but also other resources.  AFAICT dox is the former, and doesn't help with the latter.

We already had an existing solution that generated HTML; that was changed in this bug to integrate with the Pages template system.  I agree that we should use some sort of more formal templating system, so that headers/etc are consistent across all the documentation.  But I think we need to switch to a node based system we can use locally, rather than relying on Github's stuff.

> I would really like that it would be a one-page documentation, because it is that much faster to jump between and you can search the whole documentation with the native browser search.

I tend to think that would be far too unwieldy.   But maybe if I saw it in practice it would be ok?

**Comment by starwed** [comment]
Some searching turned up [assemble](http://assemble.io/), which is a node/grunt-based static site generator.  [This blurb](http://stackoverflow.com/a/16254540/460138) makes it sound ideal for our purposes, but I haven't actually looked at it in depth yet!

**Comment by kevinsimper** [comment]
Yeah, that looks cool, but we still has extract the documentation from the source files, and my problem when i just quickly tried, was that the you would have to categories the json like the build.coffee does.

If we could make the build.coffee to output json it would be pretty quick using assemble.

**Comment by starwed** [comment]
I think assemble can just take HTML as the input.  Currently the documentation looks like this, after building it:

```
--- 
layout: default
---

<div id='doc-nav'> [...]</div>
<div id='doc-content'>[...]</div>
```

When we upload that to github Pages, it takes the front matter (the bit delimited by `---`) and adds the appropriate header etc.  Assemble should be able to do the same thing locally, transforming the docs we generate without needing any extra data beyond what we already provide Pages.

[Here](https://github.com/assemble/boilerplate-site/blob/master/src/templates/pages/examples.hbs) is an example of an input file.  It should work just as well with static files as with templates.  And then we can switch to using templates later, if it seems like a good idea.  :)

_e:_ Heh, I've been trying to install `assemble`, but it turns out npmjs.org is having problems... keeps lagging out or returning server errors.

**Comment by starwed** [comment]
Now that npmjs.org is back up, I managed to install assemble and get it to transform the documentation.  It's actually fairly straightforward, though I had to tweak the layout template a tiny bit.  (They use `{{>body}}` instead of `{{content}}` for the main page content; I'm guessing there's a way around that.)

It seems like a fairly versatile tool, though their documentation is frustratingly vague in lots of places.  [This tutorial](http://www.fettblog.eu/blog/2013/09/02/using-assemble-io-with-yeoman-ios-webapp-gruntfile/) explains the basics pretty well.
